app/views/editor/verifier/i18nVerifierView.coffee

```diff
@@ -0,0 +1,156 @@
+          @percentDifference(p) < @completeThreshold and\
+          (p.count / @totalCount) >= (@countThreshold / 100))
+      .uniq((p) -> p.trimmed)
+      .value()
+    problems: ->
+      _.sortBy(_.flatten(Object.values(@problemsByLevel), true), (p) -> -p.count)
+  created: ->
+    i18n.setLng(@language)
+    @loadCampaigns()
+    @loadLanguage(@language).then =>
+      @setupRegexes()
+      @getProblems(@levelSlug).then (newProblems) =>
+        @compareStrings(newProblems)
+        @loading = false
+  watch:
+    language: ->
+      @loadLanguage(@language).then =>
+        @setupRegexes()
+        @compareStrings(@problems)
+    selectedLevelSlugs: ->
+      @loading = true
+      promises = []
+      for slug in @selectedLevelSlugs
+        if not @problemsByLevel[slug]
+          promises.push @getProblems(slug)
+      Promise.all(promises).then (newProblems) =>
+        @loading = false
+        _.defer =>
+          @compareStrings(_.flatten(newProblems))
+    messageOrHint: ->
+      @compareStrings(@problems)
+  methods:
+    loadLanguage: (language) ->
+      new Promise (accept, reject) =>
+        loading = application.moduleLoader.loadLanguage(language)
+        if loading
+          application.moduleLoader.once 'load-complete', accept
+        else
+          accept()
+    loadCampaigns: ->
+      $.get(
+        '/db/campaign',
+        (@campaigns) =>
+          @selectedCampaign = _.find(@campaigns, (c) -> c.name is "Dungeon")
+          for campaign in @campaigns
+            Vue.set(campaign, 'levelsArray', Object.values(campaign.levels))
+      )
+    setupRegexes: ->
+      en = require('locale/en').translation
+      otherLang = require("locale/#{@language}").translation
+      translationKeys = Object.keys(en.esper)
+      @regexes = []
+      for translationKey in translationKeys
+        englishString = en.esper[translationKey]
+        regex = Problem.prototype.makeTranslationRegex(englishString)
+        @regexes.push(regex)
+      @otherRegexes = []
+      for translationKey in translationKeys
+        otherString = otherLang.esper?[translationKey] or ''
+        otherRegex = Problem.prototype.makeTranslationRegex(otherString)
+        @otherRegexes.push(otherRegex)
+    percentDifference: (problem) ->
+      ((1 - problem.trimmed?.length / problem[@messageOrHint].length) * 100).toFixed(0)
+    color: (problem) ->
+      amountTranslated = @percentDifference(problem)
+      if amountTranslated >= @completeThreshold
+        return 'green'
+      else if amountTranslated >= @partialThreshold
+        return 'yellow'
+      else
+        return 'red'
+    getProblemsAndCompare: (levelSlug) ->
+      @getProblems(levelSlug).then (problems) =>
+        @compareStrings(problems)
+    getProblems: (levelSlug) ->
+      new Promise (accept, reject) =>
+        $.post(
+          '/db/user.code.problem/-/common_problems',
+          {startDay: @startDay, endDay: @endDay, slug: levelSlug},
```

Comment:
Same, add a function to `app/core/api` and you won't have to manually wrap it in a Promise.